config/generation.php

```diff
@@ -1,319 +1,59 @@
-        'safe' => true,
-    ],
-    [
-        'source' => __DIR__.'/../dist/svg/negative/symbols',
-        'destination' => __DIR__.'/../resources/svg',
-        'output-prefix' => 'n-',
-        'after' => $svgNormalization,
-        'safe' => true,
-    ],
-    [
-        'source' => __DIR__.'/../dist/svg/outline/symbols',
-        'destination' => __DIR__.'/../resources/svg',
-        'output-prefix' => 'o-',
-        'after' => $svgNormalization,
-        'safe' => true,
-    ],
-    // typography icons
-    [
-        'source' => __DIR__.'/../dist/svg/filled/typography',
-        'destination' => __DIR__.'/../resources/svg',
-        'output-prefix' => 'f-',
-        'after' => $svgNormalization,
-        'safe' => true,
-    ],
-    [
-        'source' => __DIR__.'/../dist/svg/negative/typography',
-        'destination' => __DIR__.'/../resources/svg',
-        'output-prefix' => 'n-',
-        'after' => $svgNormalization,
-        'safe' => true,
-    ],
-    [
-        'source' => __DIR__.'/../dist/svg/outline/typography',
-        'destination' => __DIR__.'/../resources/svg',
-        'output-prefix' => 'o-',
-        'after' => $svgNormalization,
-        'safe' => true,
-    ],
-    // vehicles icons
-    [
-        'source' => __DIR__.'/../dist/svg/filled/vehicles',
-        'destination' => __DIR__.'/../resources/svg',
-        'output-prefix' => 'f-',
-        'after' => $svgNormalization,
-        'safe' => true,
-    ],
-    [
-        'source' => __DIR__.'/../dist/svg/negative/vehicles',
-        'destination' => __DIR__.'/../resources/svg',
-        'output-prefix' => 'n-',
-        'after' => $svgNormalization,
-        'safe' => true,
-    ],
-    [
-        'source' => __DIR__.'/../dist/svg/outline/vehicles',
-        'destination' => __DIR__.'/../resources/svg',
-        'output-prefix' => 'o-',
-        'after' => $svgNormalization,
-        'safe' => true,
-    ],
-];
+/**
+ * Define a function to produce the config generator.
+ *
+ * Using iterator to array we can turn this into a generated version of the previous config.
+ * The benefit here is that due to how the source/upstream organizes icons by context type.
+ * But we want to output all the icons into sets based only on style type.
+ *
+ * @return Generator
+ */
+function build_generation_config(): Generator {
+    $svgNormalization = static function (string $tempFilepath, array $iconSet) {
+        /** @var string $svgContent */
+        $svgContent = file_get_contents($tempFilepath);
+        // replace <svg width="48" height="48" viewBox="0 0 48 48" fill="none" with <svg fill="currentColor" stroke="none" viewBox="0 0 24 24" that are expected by blade-icons package
+        $svgContent = str_replace('<svg width="48" height="48" viewBox="0 0 48 48" fill="none"', '<svg fill="currentColor" stroke="none" viewBox="0 0 48 48" ', $svgContent);
+        file_put_contents($tempFilepath, $svgContent);
+    };
+
+    static $upstreamTypeSets = [
```

Comment:
Moving forward if the healthicons pack source adds more of these "type based sets", it's as simple as adding the folder name here. Then the generator will update the configs final results for all 3 "style based sets" automatically.

Reply:
This is almost magical IMO